packages/core-suggest/core-suggest.js

```diff
@@ -166,10 +166,17 @@ function onAjax (self) {
 function onAjaxSend (self, ajax) {
   if (!self.input.value) return // Abort if input is empty
   if (dispatchEvent(self, 'suggest.ajax.beforeSend', ajax)) {
+    ajax.onerror = () => dispatchEvent(self, 'suggest.ajax.error', ajax)
     ajax.onload = () => {
-      try { ajax.responseJSON = JSON.parse(ajax.responseText) } catch (err) { ajax.responseJSON = false }
+      if (ajax.status !== 200) return dispatchEvent(self, 'suggest.ajax.error', ajax)
+      try {
+        ajax.responseJSON = JSON.parse(ajax.responseText)
+      } catch (err) {
+        ajax.responseJSON = false
+        const detail = Object.assign({}, ajax, { statusText: err.toString() }) // Don't modify global ajax status text
```

Comment:
Maybe we should rather have a ajax-object for each instance? and modify this (instead of extending an object, so the consumer have a consistent access to the ajax-object and can inspect it further as well as use methods?) :)

Timeout can be implemented by using a setTimeout, and an prop/attribute to change standard (no timeout), but this can be another task I guess :)

Reply:
> Maybe we should rather have a ajax-object for each instance? and modify this (instead of extending an object, so the consumer have a consistent access to the ajax-object and can inspect it further as well as use methods?) :)

do you mean to create a new xhr object for each ajax call?

Reply:
ah, read only.. could we make a custom prop like we do with `responseJSON`? :)
was thinking a xhr-object pr. element instance, not pr. event (to still be able to do `abort()`) :)